lib/matplotlib/widgets.py

```diff
@@ -2206,7 +2240,11 @@ def _setup_edge_handle(self, props):
                                              direction=self.direction,
                                              line_props=props,
                                              useblit=self.useblit)
-        self.artists.extend([line for line in self._edge_handles.artists])
+
+    @property
+    def _handles_artists(self):
+        if self._edge_handles is not None:
+            return self._edge_handles.artists
```

Comment:
We should have type stability, i.e.  `_handles_artists` should always return a tuple.
```suggestion
            return self._edge_handles.artists
        else:
            return ()
```